### PR TITLE
fix(github-embed-23055): resolve relative GitHub README image URLs

### LINKED ISSUE #23055: {% github %} liquid tag: relative image paths in README resolve to broken URLs and get permanently cached
**Describe the bug**

The `{% github %}` liquid tag embeds a GitHub repository's README, but relative image paths (e.g. `screenshot`) are resolved against the repository's HTML web URL (`https://github.com/owner/repo/docs/screenshot.png`). GitHub does not serve raw image bytes from that path — it returns an HTML page — so the images render broken. The CDN then caches that failed response, and because the cache key is the resolved URL, the broken image is nearly impossible to recover without direct cache manipulation.

**To Reproduce**

1. Find or create a public GitHub repo whose README contains at least one relative image path (not an absolute `https://` URL).
2. In a Forem post, add `{% github https://github.com/owner/repo %}`.
3. Publish or preview the post.
4. Observe that README image(s) render as broken/missing.
5. Note that even after fixing the upstream image, the broken result persists due to CDN caching of the invalid response.

**Expected behavior**

Relative `img src` paths should be rewritten to their raw content equivalent before rendering — e.g. `https://raw.githubusercontent.com/owner/repo/HEAD/docs/screenshot.png` — so the CDN caches a valid image response instead of a broken one.

**Screenshots**

Reproducible with any public repo using relative image paths in its README (e.g. many popular open source projects).

**Desktop (please complete the following information):**

- OS, version: N/A (server-side render)
- Browser, version: N/A

**Smartphone (please complete the following information):**

- Device: N/A
- OS, version: N/A
- Browser, version: N/A

**Additional context**

Root cause is in `GithubTag::GithubReadmeTag#clean_relative_path!` (`app/liquid_tags/github_tag/github_readme_tag.rb`). The method resolves all relative paths (both `img src` and `a href`) against `content.html_url`. For `img` elements, the correct target is `https://raw.githubusercontent.com/{owner}/{repo}/{default_branch}/{path}`. The `default_branch` is already available on the `content` object returned by the existing `Github::OauthClient.new.repository(repository_path)` call — no additional API calls are needed.

### PR DESCRIPTION
**Pull request by anchildress1**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Fix relative image paths in GitHub embed rendering
- Add tests to verify correct img src rewriting
- Add tests to verify non-rewriting for absolute/root-relative paths

## Related Tickets & Documents

- Closes #23055 

## QA Instructions, Screenshots, Recordings

_Follow better instructions in issue #23055 to reproduce._
_Tested with Chrome Dev._
_Reproducible with any GitHub embed having relative img in README._
_Create post > paste gh embed > observe._

### Before
Screenshot 2026-03-30 at 10 33 35 PM

### After
Screenshot 2026-03-30 at 10 34 34 PM

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?
Nope. Pretty straightforward cure for my OCD.

## [optional] What gif best describes this PR or how it makes you feel?

this-is-fine-its-fine
Really... but what do you guys think about using OG instead of the `README` for the GH embed?? 😆

### CODE REVIEW

**Comment by anchildress1** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/liquid_tags/github_tag/github_readme_tag.rb [hunk 1]
```diff
@@ -97,6 +97,9 @@ def clean_relative_path!(readme_html, url)
         elsif path.start_with?("#")
           # Anchor link (e.g., #license)
           element.attributes[attribute].value = "#{url}#{path}"
+        elsif element.name == "img"
+          # Relative image path — resolve to raw content URL so the CDN caches valid bytes
+          element.attributes[attribute].value = "#{url.sub('github.com', 'raw.githubusercontent.com')}/HEAD/#{path}"
```
[the remark this answers is not in the document]
Comment (anchildress1) [reply app/liquid_tags/github_tag/github_readme_tag.rb:103]: Fixed with additional skip logic.

File: spec/liquid_tags/github_tag/github_readme_tag_spec.rb [hunk 2]
```diff
@@ -128,9 +128,42 @@ def generate_tag(path, options = "")
...
+        expect(result).to include('src="https://github.com/owner/repo/raw/main/logo.png"')
+        expect(result).not_to include("raw.githubusercontent.com")
+      end
+
+      it "does not rewrite a relative a href to raw.githubusercontent.com" do
+        html = '<a href="blob/main/README.md">README</a>'
+        result = tag.send(:clean_relative_path!, html, repo_url)
+        expect(result).to include('href="https://github.com/owner/repo/blob/main/README.md"')
+        expect(result).not_to include("raw.githubusercontent.com")
+      end
```
[the remark this answers is not in the document]
Comment (anchildress1) [reply spec/liquid_tags/github_tag/github_readme_tag_spec.rb:166]: Added missing edge case tests.
